### PR TITLE
Log traceback for failures in submitting task

### PR DESCRIPTION
**Pull request by alexrson**
Currently, when there's an error in the process of submitting a task we only log what the error is. This adds the traceback.

## Motivation and Context

I was trying to find a bug in my pipeline code with only the hint that it was `'dict' object is not callable`. So i had to add print statements on every line in several classes until i found it.

But with this change, one would be able to tell which exact line this bug was on. 

## Have you tested this? If so, how?
 
Basically no, but I would really like to, but I'm not sure how, since this is my first commit to this repo and I'm not sure how to get it running for real.

### CODE REVIEW

File: luigi/worker.py [hunk 1]
```diff
@@ -1006,6 +1006,8 @@ def _run_async_task(self, task):
             task.submit_task()
         except Exception as ex:
             logger.error("Error in submitting task: %s %s", task.task_id, ex)
```
Comment (abhik) [inline luigi/worker.py]: Can you check if replacing with `logging.exception` leads to the same output? It should.

Comment (alexrson) [reply luigi/worker.py]: Yes, I changed it to do that instead.
